packages/provider/src/utils.ts

```diff
@@ -111,8 +111,8 @@ export const isWalletUpToDate = (status: AccountStatus): boolean => {
 }
 
 export interface ItemStore {
-  getItem(key: string): Promise<string | null>
```

Comment:
This store is used dapp-side, so I figure 100% of integrations are using the default (local storage). Making this async is a bit problematic, because ethers expects the provider to know the "initial network" in a sync way.

I say it's better to leave it like this for now, and if we want to support different stores later we can revisit it.